ddtrace/contrib/pytest/plugin.py

```diff
@@ -86,48 +82,31 @@ def pytest_addoption(parser):
 
 def pytest_configure(config):
     config.addinivalue_line("markers", "dd_tags(**kwargs): add tags to current span")
-    if is_enabled(config):
-        ci_tags = ci.tags()
-        if ci_tags.get(ci.git.REPOSITORY_URL, None) and int_service(None, ddtrace.config.pytest) == "pytest":
-            repository_name = _extract_repository_name(ci_tags[ci.git.REPOSITORY_URL])
-            ddtrace.config.pytest["service"] = repository_name
-
-        try:
-            from ddtrace.internal.codeowners import Codeowners
-
-            ddtrace.config.pytest["_codeowners"] = Codeowners()
-        except ValueError:
-            # the CODEOWNERS file is not available
-            pass
-        except Exception:
-            log.warning("Failed to load CODEOWNERS", exc_info=True)
-
-        Pin(tags=ci_tags, _config=ddtrace.config.pytest).onto(config)
 
 
 def pytest_sessionstart(session):
-    pin = Pin.get_from(session.config)
-    if pin is not None:
-        tracer_filters = pin.tracer._filters
-        if not any(isinstance(tracer_filter, TraceCiVisibilityFilter) for tracer_filter in tracer_filters):
-            tracer_filters += [TraceCiVisibilityFilter()]
-            pin.tracer.configure(settings={"FILTERS": tracer_filters})
+    if is_enabled(session.config):
+        _CIVisibility.enable(config=ddtrace.config.pytest)
```

Comment:
From the PR description:

> The Pin class is traditionally used for integrations, but since CIVisibility is meant to be run only for a test suite in CI, the corresponding tracer will only be configured for that specific test suite and shutdown once the test suite is finished.

Note that we also want to support sending events for "modules" (groups of test suites) and "sessions" (the whole execution of a test command), as part of the Test Suite Level Visibility task. From your wording in the PR description it seems it will be harder to do that.

This change of moving the initialization to `pytest_sessionstart` also seems to be something we will need to change then.

Is this something we should take into account now?

Reply:
Good point, I think I used the wrong wording in that description. I meant that since CIVisibility is meant to run only for the execution of a test session (likely a CI job), the corresponding tracer will only need to be configured for each test session and then be shutdown at the end of the session. Hopefully this clarifies any of the confusion, and I think this shouldn't change our plan of supporting module/session events.

I don't think there's a huge difference between initializing at `pytest_configure` vs `pytest_sessionstart` since the latter is executed almost immediately after the former, but since `pytest_configure` is closer to the start of the actual test command I can move the initialization back to `pytest_configure`.